lib/s3_meta_sync/syncer.rb

```diff
@@ -83,6 +83,18 @@ def download(source, destination)
       end
     end
 
+    def make_temporary_folder
+      dir = Dir.mktmpdir("s3ms_")
+      log "Tmp folder: #{dir}"
+      yield dir
+    ensure
+      if dir
+        log "FileUtils.remove_entry #{dir}: #{FileUtils.remove_entry(dir, true)}"
+        # Sometimes remove_entry fails, trying really hard:
+        log "rm -rf #{dir}: #{`rm -rf #{dir}`.inspect}"
+      end
+    end
```

Comment:
Prefer to keep mktmpdir and only do cleanup and log if something went wrong ... we should trust ruby to work by default ...